src/Sylius/Bundle/AddressingBundle/DependencyInjection/Configuration.php

```diff
@@ -77,7 +81,7 @@ private function addResourcesSection(ArrayNodeDefinition $node): void
                                         ->scalarNode('model')->defaultValue(Address::class)->cannotBeEmpty()->end()
                                         ->scalarNode('interface')->defaultValue(AddressInterface::class)->cannotBeEmpty()->end()
                                         ->scalarNode('controller')->defaultValue(ResourceController::class)->cannotBeEmpty()->end()
-                                        ->scalarNode('repository')->cannotBeEmpty()->end()
+                                        ->scalarNode('repository')->defaultValue(AddressRepository::class)->cannotBeEmpty()->end()
```

Comment:
Sylius doesn't specify repository classes in DI configuration up to 1.12 (not a single case, even if repository classes existed), cases started to appear in 1.13 and in catalog promotion source code.

I'm unaware of the reason or code mechanics behind it, but let's find correct solution and either specify repository classes for all bundles or for none as it were in 1.12 and below. 

@jakubtobiasz we discussed it in a other PR, WDYT?

Requesting changes since this is a prime PR to solve the mystery.

Reply:
Hi @diimpp 👋🏼,
this task is a result of #15275, where we added a repository, but if anyone already created a new repository, they won't be able to use our new one (custom one extends `EntityRepository` instead of Sylius' `ZoneRepository`).

@Rafikooo split this one big task into four smaller ones, and this is the first one. We'll unify this thing across all bundles :).